code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,865 @@
+
+
+
+/obj/effect/warped_rune/pinkspace
+	desc = "Love is the only reliable source of happiness we have left. But like everything, it comes with a price."
+	icon_state = "love_rune"
+
+
+///adds the hug mood effect along with the sound effect.
+/obj/effect/warped_rune/pinkspace/Crossed(atom/movable/AM)
+	if(istype(AM,/mob/living/carbon/human))
+		playsound(T, "sound/weapons/thudswoosh.ogg", 50, TRUE)
+		SEND_SIGNAL(AM, COMSIG_ADD_MOOD_EVENT,"jolly", /datum/mood_event/jolly)
+		AM.visible_message(AM, "<span class='notice'>You start feeling happier somehow.</span>")
+
+
+
+
+/*Gold rune : Turn things over it to gold, completely fucks over economy */
+
+/obj/item/slimecross/warping/gold
+	colour = "gold"
+	runepath = /obj/effect/warped_rune/goldspace
+	effect_desc = "Draws a rune capable of turning nearly anything put on it to gold."
+
+/obj/effect/warped_rune/goldspace
+	icon_state = "midas_rune"
+	desc = "When everything is abundant, it becomes easy to forget when one had to work to obtain anything."
+
+/obj/effect/warped_rune/goldspace/Initialize()
+	.=..()
+	RegisterSignal(get_turf(src),COMSIG_ATOM_ENTERED,.proc/midas_touch)
+
+///turn anything on the tile to a golden version of themselves. Sheets of any kind will be directly turned into gold bars.
+/obj/effect/warped_rune/goldspace/proc/midas_touch()
+	for(var/obj/item/stack/sheet/S in T)
+		if(istype(S,/obj/item/stack/sheet/mineral/gold)) //can't turn gold into gold
+			return
+		var/obj/item/stack/sheet/mineral/gold/G = new(T)
+		G.amount = S.amount //as broken as it sounds if you mass produce easy stacks
+		qdel(S)
+		return
+
+	for(var/obj/item/I in T)
+		I.material_flags = MATERIAL_COLOR | MATERIAL_ADD_PREFIX | MATERIAL_AFFECT_STATISTICS
+		I.set_custom_materials(list(/datum/material/gold=500)) //always set at 500 to not make it too abusable
+
+
+/*Adamantine rune, will spawn ores depending on the mineral rocks surrounding it. Here to make miners do their job even less.  */
+
+/obj/item/slimecross/warping/adamantine
+	colour = "adamantine"
+	runepath = /obj/effect/warped_rune/adamantinespace
+	effect_desc = "draws a rune capable of copying the ores of nearby mineral rocks."
+
+/obj/effect/warped_rune/adamantinespace  //doesn't have a rune icon yet please spriters help me I can't sprite for shit I beg you
+	desc = "The universe's ressource are nothing but tools for us to use and abuse."
+	max_cooldown = 300  //"mines" things every 30 seconds.
+
+/obj/effect/warped_rune/adamantinespace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+/obj/effect/warped_rune/adamantinespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		auto_mining()
+
+/obj/effect/warped_rune/adamantinespace/proc/auto_mining()
+	for(var/turf/closed/mineral/M in range(7,T)) //the range is pretty big to at least try to rival miners and their plasma cutters.
+		if(M.mineralType == null) //here to counter runtimes when the mineral type of the rock is null, does absolutely nothing.
```

Comment:
can't you just use if(M.mineralType!=null) here and remove the else?

Reply:
not really, if we don't use else it will use the next code in line, aka the one we're trying to not use, and using an early return will stop the rune BEFORE it gets to duplicate every rune in range.